File: movielens/ml_analysis.rb
require "jubatus/recommender/client"
require "jubatus/recommender/types"
require "pp"

recommender = Jubatus::Recommender::Client::Recommender.new "127.0.0.1", 9199

Array.new(943){|index| "#{index}"}.each{|n|
  sr = recommender.similar_row_from_id("movie_len", n.to_s, 10)
  print "user#{n} is similar to "
  sr.each{|user_tuple|
    print(" user#{user_tuple[0]} score:#{user_tuple[1]}  ")
  }
  puts ""
}

File: movielens/python/ml_analysis.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys
from jubatus.recommender import client
from jubatus.recommender import types

if __name__ == '__main__':

    recommender = client.Recommender("127.0.0.1", 9199, 'movie_len')

    for i in range(1,943):
        sr = recommender.similar_row_from_id(str(i) , 10);
        print("user ", str(i),  " is similar to :", list(map(str, sr)))
  
